framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -285,104 +283,101 @@ protected boolean areEqualInHierarchy(
   }
 
   /**
-   * A declared type is considered a supertype of another declared type only if all of the type
-   * arguments of the declared type "contain" the corresponding type arguments of the subtype.
-   * Containment is described in <a
+   * Returns true if {@code outside} contains {@code inside}, that is, if the set of types denoted
+   * by {@code outside} is a superset of or equal to the set of types denoted by {@code inside}.
+   *
+   * <p>Containment is described in <a
    * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-4.html#jls-4.5.1">JLS section
-   * 4.5.1 "Type Arguments of Parameterized Types"</a>.
+   * 4.5.1 "Type Arguments of Parameterized Types"</a>. A declared type is considered a supertype of
+   * another declared type only if all of the type arguments of the declared type "contain" the
+   * corresponding type arguments of the subtype.
    *
-   * @param inside a type argument of the "subtype"
-   * @param outside a type argument of the "supertype"
+   * @param inside a possibly-contained type
+   * @param outside a possibly-containing type
    * @param canBeCovariant whether or not type arguments are allowed to be covariant
    * @return true if inside is contained by outside, or if canBeCovariant == true and {@code inside
    *     <: outside}
    */
   protected boolean isContainedBy(
-      final AnnotatedTypeMirror inside, final AnnotatedTypeMirror outside, boolean canBeCovariant) {
+      AnnotatedTypeMirror inside, AnnotatedTypeMirror outside, boolean canBeCovariant) {
 
     if (ignoreUninferredTypeArgument(inside) || ignoreUninferredTypeArgument(outside)) {
       areEqualVisitHistory.put(inside, outside, currentTop, true);
       return true;
     }
-
-    if (outside.getKind() != TypeKind.WILDCARD
-        && !TypesUtils.isCaptured(outside.getUnderlyingType())) {
-      if (canBeCovariant) {
-        return isSubtype(inside, outside, currentTop);
+    if (outside.getKind() == TypeKind.WILDCARD
+        // TODO: the below should be removed after #979 is fixed.
+        || (TypesUtils.isCapturedTypeVariable(outside.getUnderlyingType())
+            && !TypesUtils.isCapturedTypeVariable(inside.getUnderlyingType()))) {
```

Comment:
I don't understand this line.  (I understand the one above, but I don't see why this part of the test is desirable/necessary.)  Maybe it doesn't matter because this is a hack to be removed later.

Reply:
I added a comment.